internal/cli/project_apply.go

```diff
@@ -254,6 +255,22 @@ func (c *ProjectApplyCommand) Run(args []string) int {
 		)
 	}
 
+	// Setup our default waypoint.hcl if it was given
+	if v := c.flagWaypointHcl; v != "" {
+		bs, err := ioutil.ReadFile(v)
+		if err != nil {
+			c.ui.Output(
+				"Error reading HCL file specified with the -waypoint-hcl flag: %s",
+				clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+
+			return 1
+		}
+
+		proj.WaypointHcl = bs
```

Comment:
Do we do any simple HCL syntax check at all before sending it off to the server, or does the CLI just assume it's ok?

Reply:
A syntax check is a good idea. 

I didn't want to do a full parse cause if a user has like a `file()` call that depends on the source, it will fail. But checking actually syntax only (basic parse) is a good idea.

Reply:
Done!